src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -189,10 +212,50 @@ private[impl] class InstanceTrackerActor(
     }
 
     instancesBySpec = updatedAppInstances
+    println(instancesBySpec)
     counts = updatedCounts
 
     // this is run on any state change
     metrics.stagedCount.setValue(counts.tasksStaged.toLong)
     metrics.runningCount.setValue(counts.tasksRunning.toLong)
   }
+
+  /**
+    * If we encounter failure, we try to reload the effected task to make sure that the taskTracker
+    * is up-to-date. We signal failure to the sender if the state is not as expected.
+    *
+    * If reloading the tasks also fails, the operation does fail.
+    *
+    * This tries to isolate failures that only effect certain tasks, e.g. errors in the serialization logic
+    * which are only triggered for a certain combination of fields.
+    */
+  private def tryToRecoverRepositoryFailure(update: InstanceUpdateEffect.Update)(
```

Comment:
I'd say the master should abdicate. However, since this a change in the behaviur we should not hide it in this diff but make a separate diff. What do you think, @meichstedt and @timcharper?

Reply:
Ideally, this logic goes into the failure handling of the `pipe` call (see my comment above).

Reply:
I've talked to @meichstedt. He said we should just abdicate when we fail to write something.